ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,295 @@
+        .format(elb_name))
+    if 'elb_name' in ctx.instance.runtime_properties:
+        ctx.instance.runtime_properties.pop('elb_name')
+
+    return
+
+
+def _create_health_check(user_health_check):
+
+    health_check = {
+        'interval': constants.HEALTH_CHECK_INTERVAL,
+        'healthy_threshold': constants.HEALTH_CHECK_HEALTHY_THRESHOLD,
+        'timeout': constants.HEALTH_CHECK_TIMEOUT,
+        'unhealthy_threshold': constants.HEALTH_CHECK_UNHEALTHY_THRESHOLD
+    }
+
+    health_check.update(user_health_check)
+
+    health_check = HealthCheck(**health_check)
+    return health_check
+
+
+def _create_elb_params():
+    params_dict = {'listeners': ctx.node.properties['listeners'],
+                   'name': ctx.node.properties['elb_name'],
+                   'zones': ctx.node.properties['zones']}
+    if 'security_groups' in ctx.node.properties.keys():
+        params_dict['security_groups'] = ctx.node.properties['security_groups']
+    return params_dict
+
+
+def _create_elb():
+
+    ctx.logger.info('Attempting to Create Load Balancer.')
+
+    params_dict = _create_elb_params()
+    elb_client = connection.ELBConnectionClient().client()
+
+    try:
+        lb = elb_client.create_load_balancer(**params_dict)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer not created '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info('Load Balancer Created. ')
+
+    utils.set_external_resource_id(params_dict['name'],
+                                   ctx.instance,
+                                   False)
+
+    ctx.instance.runtime_properties['elb_name'] = params_dict['name']
+
+    if not lb:
+        raise NonRecoverableError('Load Balancer not created')
+
+    return lb
+
+
+def _get_instance_list():
+
+    ctx.logger.info('Attempting to get Load Balancer Instance List.')
+
+    elb_name = ctx.node.properties['elb_name']
+    lb = _get_existing_elb(elb_name)
+    list_of_instances = lb.instances
+    ctx.instance.runtime_properties[constants.EXTERNAL_RESOURCE_ID] = lb.name
+
+    return list_of_instances
+
+
+def _get_elbs_by_names(list_of_names):
+
+    ctx.logger.info(
+        'Attempting to get Load Balancers {0}.'.format(list_of_names))
+    elb_client = connection.ELBConnectionClient().client()
+
+    try:
+        elb_list = elb_client.get_all_load_balancers(
+            load_balancer_names=list_of_names)
+        total_elb_list = elb_client.get_all_load_balancers()
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Error when accessing ELB interface '
+                                  '{0}'.format(str(e)))
+
+    if not elb_list:
+        ctx.logger.error('Unable to find load balancers matching: '
```

Comment:
I understand.. but I'm actually asking whether this is always an error. Might it be that you didn't find a load balancer but it's ok that you didn't? If there is such a case, this is a "logger.error"..